docs/DEVELOPMENT-GUIDE.md

```diff
@@ -226,6 +226,24 @@ Unit tests are run automatically with `make`. To run a subset of the unit tests,
 UNIT_TESTS=s2n_hash_test make
 ```
 
+### Debugging With GDB
+When trying to debug a failing test case, it is often useful to use a debugger like `gdb`. To do this, first make sure that the tests and s2n are compiled with debug information. This can be done by setting the CMAKE_BUILD_TYPE to DEBUG. Alternately, you can set the build type to R.
+```
+# generate the build configuration with debug symbols enabled
+cmake . \
+    -Bbuild \
+    -DCMAKE_BUILD_TYPE=DEBUG
+```
+
+Then a unit test can be run from gdb. Note that our unit tests rely on relative paths for cert locations and things, so the test executable must be invoked from the directly that holds the test source file.
+
+consider the `s2n_x509_validator_test`. This source file for this resides at `s2n-tls/tests/unit/s2n_x509_validator_test.c`, while the test executable will reside at `s2n-tls/build/bin/s2n_x509_validator_test.c`. So running the test with gdb can be done with the following
```

Comment:
Do you think this example is necessary? I think the gdb description is sufficient.